### DISCUSSION TITLE
Power consumption nRF52

### DISCUSSION THREAD

**Discussion by StevenGhe** [zephyrproject-rtos/zephyr#35463] [Q&A]
Hi!

I'm a student asking about the power features enabled by Zephyr of the Nordic nRF52. 

I know there are some other nordic power modes such as System OFF, RAM power control features and such.
Is there a reason there is only support for "PM_STATE_SOFT_OFF" which leads to a 0.8 mA current? Following some advice I found a way of retaining data on setting back to "ACTIVE" state but this 800uA is still quite a lot.

Given the fact that it's not supported yet gives me the feeling that this is more than linking the API based on Zephyr's power state.

Any advice getting the power consumption down, or disabling certain features on the nRF52 SoC like the antenna etc?

Thanks

**Comment by nunof** [comment]
Hi!

I don’t have an answer for you but I want to add my case which relates to yours. Perhaps it will help trigger some answer. 

I have a nRF52832 board and I have been unable to make the sample in boards/nrf/system_off the way is intended, meaning that I still measure very high power consumption - more than 3.5mA without debugger and more 4.6mA with. Checked it’s not a problem with board or with measurement method.

It seems PM doesn’t kick in and there’s a kernel crash when entering state_off

I get this crash with debugger attached which I believe it's expected

```
ASSERTION FAIL [esf != ((void *)0)] @ ../zephyr/arch/arm/core/aarch32/cortex_m/fault.c:993

  ESF could not be retrieved successfully. Shall never occur.

[00:00:02.425,170] <err> os: r0/a1:  0x00000004  r1/a2:  0x000003e1  r2/a3:  0x00000000
[00:00:02.425,201] <err> os: r3/a4:  0x00000007 r12/ip:  0x2e727563 r14/lr:  0x000048df
[00:00:02.425,201] <err> os:  xpsr:  0x61000000
[00:00:02.425,201] <err> os: Faulting instruction address (r15/pc): 0x0001d2b8
[00:00:02.425,201] <err> os: >>> ZEPHYR FATAL ERROR 4: Kernel panic on CPU 0
[00:00:02.425,231] <err> os: Current thread: 0x20001518 (unknown)
[00:00:03.856,475] <err> os: Halting system
```

Without debugger I can't capture any message but I can still see the high power consumption

Thoughts?
